classes/fetch_services.py

```diff
@@ -124,16 +128,47 @@ def parse(self, response) -> dict:
         upload_date = datetime.strptime(response.get("upload_date"), date_format)
         upload_date = upload_date.replace(tzinfo=timezone.utc)
 
+        if response.get("duration") is None and prompt_dur:
+            mins = None
+            seconds = None
+            err("Missing Duration, Please Add Manually:")
+
+            while(True):
+                try:
+                    if mins is None:
+                        mins = int(input("Minutes: "))
```

Comment:
It's a useful feature, although the user will have to realize that they need to do this in the console, otherwise they will be sitting at the application window wondering why it's stopped. :) Still, since they have to opt-in to the feature via checkbox, it's not too much of a problem.

Reply:
Just above this I also have the print statement  `err("Missing Duration, Please Add Manually:")` to kind of indicate why it stopped